lib/teams/index.ts

```diff
@@ -294,3 +298,185 @@ export class PlatformTeam extends ApplicationTeam {
+   * This method 
+   * @param cluster EKS cluster where to apply the RBAC
+   * @param namespace Namespace where the RBAC are applied
+   * @param createNamespace flag to create namespace if not already created
+   * @returns 
+   */
+
+  private setEmrContainersForNamespace(cluster: Cluster, namespace: string, createNamespace: boolean): KubernetesManifest {
+
+    //Get the Role definition and add the namespace of the EMR on EKS virtual cluster
+    let emrContainersK8sRole = readYamlDocument(`${__dirname}/emrContainersRole.yaml`);
+    emrContainersK8sRole = emrContainersK8sRole.replace('<REPLACE-NAMESPACE>', namespace);
+    const emrContainersK8sRoleManifest = loadYaml(emrContainersK8sRole);
+
+    //Create the role used by EMR on EKS
+    const emrContainersK8sRoleResource = cluster.addManifest('emrContainersK8sRoleManifest',
+      emrContainersK8sRoleManifest
+    );
+
+    //Get the RoleBinding definition and add the namespace of the EMR on EKS virtual cluster  
+    let emrContainersK8sRoleBinding = readYamlDocument(`${__dirname}/emrContainersRoleBinding.yaml`);
+    emrContainersK8sRoleBinding = emrContainersK8sRoleBinding.replace('<REPLACE-NAMESPACE>', namespace);
+    const emrContainersK8sRoleBindingManifest = loadYaml(emrContainersK8sRoleBinding);
+
+    //Create the role binding between the service account and the role to be used by EMR on EKS
+    const emrContainersK8sRoleBindingResource = cluster.addManifest('emrContainersK8sRoleBindingManifest',
+      emrContainersK8sRoleBindingManifest
+    );
+
+    emrContainersK8sRoleBindingResource.node.addDependency(emrContainersK8sRoleResource);
+
+    if (createNamespace) {
+      const namespaceManifest = NameSpaceUtil.createNamespace(namespace, cluster, true);
+      emrContainersK8sRoleResource.node.addDependency(namespaceManifest);
+    }
+
+    return emrContainersK8sRoleBindingResource;
+  }
+
+  /**
+   * @internal
+   * private method to create the execution role for EMR on EKS scoped to a namespace and a given IAM role
+   * @param cluster EKS cluster
+   * @param policy IAM policy to use with the role
+   * @param namespace Namespace of the EMR Virtual Cluster
+   * @param name Name of the IAM role
+   * @returns Role
+   */
+  private createExecutionRole(cluster: Cluster, policy: IManagedPolicy, namespace: string, name: string): Role {
+
+    const stack = cluster.stack;
+
+    let irsaConditionkey: CfnJson = new CfnJson(stack, `${name}roleIrsaConditionkey'`, {
+      value: {
+        [`${cluster.openIdConnectProvider.openIdConnectProviderIssuer}:sub`]: 'system:serviceaccount:' + namespace + ':emr-containers-sa-*-*-' + Aws.ACCOUNT_ID.toString() + '-' + SimpleBase.base36.encode(name),
+      },
+    });
+
+    // Create an execution role assumable by EKS OIDC provider and scoped to the service account of the virtual cluster
+    return new Role(stack, `${name}ExecutionRole`, {
```

Comment:
I assume it is likely that a customer may use multiple clusters in the same account. Given that role name is unique globally it would be quite inconvenient to leverage the team in this setup. If you don't pass the roleName CDK will create a unique name for you. Do you see an issue with that?

Reply:
I need to have the `roleName`, its a required parameter in interface `ExcutionRoleDefinition`